the-collab-lab/tcl-22-smart-shopping-list
# Log New Tokens in Firestore Immediately After Creation

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



When a new token is created and nothing is added to the Grocery List, the token isn't added as a collection to Firestore. The collection is only added when the first shopping item is added to the list.

I couldn't find a way to create an empty collection in Firestore, so I added a document that is not a part of the grocery list.  It is called 'ListData' and stores a date when the list was created. Adding this document automatically creates a collection under the new token. (If there are more efficient solutions out there, please let me know.)

## Related Issue


Closes #26 

## Acceptance Criteria



* New token is added to Firestore immediately upon creation
* Grocery List and Add Items functionality is intact

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |

## Updates

### Before




### After


https://www.loom.com/share/d62adcf114c2422e9faf0ff2d85aabdc

## Testing Steps / QA Criteria



To test the feature, use one of the following ways:

1. Follow the [preview link](https://tcl-22-shopping-list--pr56-dt-new-token-firesto-1w6ufd54.web.app)
2. Launch the app locally from your terminal:
* Pull down this branch with `git pull origin dt-new-token-firestore` and check that branch out with `git checkout origin dt-new-token-firestore`
* Run `npm start`  to launch the app in your browser at [http://localhost:3000](http://localhost:3000)

When on application:

1. Clear your localStorage
2. Create a new list
3. Check your token value in localStorage
4. Open [Firebase](https://console.firebase.google.com/u/2/project/tcl-22-shopping-list/firestore/). Your token should be there, with one document under it called ListData.
5. Adding items and viewing your grocery list should both work the same way as before.
